cadams8238/sqsmv
# Support AWS Multifactor Authentication with Token Provider

This branch adds support for AWS MFA with a token provider (like Google Authenticator, Duo, etc).

**How to test these changes:**
* I've created 2 test queues: `sqsmv-test` and `sqsmv-test-2` that can be accessed using the `con-engineering` AWS profile (feel free to ping me if you need clarification on which account/role this profile contains)
  * `sqsmv-test` [queue](https://console.aws.amazon.com/sqs/v2/home?region=us-east-1#/queues/https%3A%2F%2Fsqs.us-east-1.amazonaws.com%2F461586521548%2Fsqsmv-test/send-receive#/)
  * `sqsmv-test-2` [queue](https://console.aws.amazon.com/sqs/v2/home?region=us-east-1#/queues/https%3A%2F%2Fsqs.us-east-1.amazonaws.com%2F461586521548%2Fsqsmv-test-2/send-receive#/)
* There is already a test message in one of those queues mentioned above, ready to be moved to the other queue for testing.
* To test the MFA changes:
  * In your terminal, export your `AWS_PROFILE` variable (ie: `export AWS_PROFILE=con-engineering` - feel free to ping me if you need to know the account/role if yours is named differently). 
  * Make sure you have MFA setup in your AWS profile
    * Check out step 5 in this [doc](https://confluence.apptio.com/pages/viewpage.action?spaceKey=ENG&title=AWS+Accounts+and+Role+Switching) if you're unsure how to do that
    * To find your MFA device arn (or to setup for the first time), see this [doc](https://confluence.apptio.com/pages/viewpage.action?spaceKey=ENG&title=How+to+Set+Up+MFA+for+Your+AWS+Account) for more details
  * If your `AWS_REGION` is not set, you'll want to export that as well (ie: `export AWS_REGION=us-east-1`).
  * Clone this fork and run the sqsmv command with the MFA flag locally, like so: `go run . -src <source queue URL> -dest <destination queue URL> -mfa`
